src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -94,11 +95,63 @@ public ProgrammingExerciseExportService(ProgrammingExerciseRepository programmin
         this.urlService = urlService;
     }
 
+    /**
+     * Exports a repository available for an instructor/tutor for a given programming exercise. This can be a template,
+     * solution, or tests repository
+     * @param exercise The programming exercise that has the repository
+     * @param repositoryType the type of repository to export
+     * @return a zipped file
+     */
+    public File exportInstructorRepositoryForExercise(ProgrammingExercise exercise, RepositoryType repositoryType) {
```

Comment:
I think there is quite some duplicated code here. Can you try to unify this with the other export functionality?

Reply:
I refactored a bit and hopefully the code is a bit better now.